src/megachatapi_impl.cpp

```diff
@@ -692,6 +692,62 @@ void MegaChatApiImpl::sendPendingRequests()
             });
             break;
         }
+        case MegaChatRequest::TYPE_PREVIEW_CHAT_LINK:
+        {
+            const char *link = request->getText();
+            string parsedLink = link;
+
+            size_t pos = parsedLink.find("#chat_");
+            if (pos == string::npos)
+            {
+                errorCode = MegaChatError::ERROR_ARGS;
+                break;
+            }
+
+            if (parsedLink.length() !=  pos + 6 + 8 + 25 + 1)
+            {
+                errorCode = MegaChatError::ERROR_ARGS;
+                break;
+            }
+
+            pos += 7;
+            string phstr = parsedLink.substr(pos, 8);
+            MegaChatHandle ph = UNDEF;
+            Base64::atob(phstr.c_str(), (byte*)&ph, MegaClient::CHATLINKHANDLE);
+
+            pos += 8;
+            string key;
+            string keystr = parsedLink.substr(pos);
+            Base64::atob(keystr, key);
+
+            if (ISUNDEF(ph) || key.length() != 16)
+            {
+                errorCode = MegaChatError::ERROR_ARGS;
+                break;
+            }
+
+            mClient->openChatLink(ph, key)
+            .then([request, this, ph]()
+            {
+                MegaChatHandle openChatId = mClient->chatIdByPh(ph);
+                request->setChatHandle(openChatId);
+
+                auto itChat = mClient->chats->find(openChatId);
```

Comment:
We have `MegaChatApiImpl::findChatRoom()` exactly for this purpose :) I will replace this code by a call to such function.